packages/report-server/src/reportBuilder/transform/aliases/keyValueByFieldAliases.ts

```diff
@@ -3,31 +3,37 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { buildTransform } from '..';
+import { Row } from '../../types';
 
-export const keyValueByDataElementName = () =>
-  buildTransform([
-    {
-      transform: 'select',
-      '$row.dataElement': '$row.value',
-      '...': ['period', 'organisationUnit'],
-    },
-  ]);
+/**
+ * { period: '20200101', organisationUnit: 'TO', dataElement: 'CH01', value: 7 }
+ *  => { period: '20200101', organisationUnit: 'TO', CH01: 7 }
+ */
+export const keyValueByDataElementName = () => (rows: Row[]) => {
+  return rows.map(row => {
+    const { dataElement, value, ...restOfRow } = row;
+    return { [dataElement as string]: value, ...restOfRow };
+  });
+};
```

Comment:
Switching from the `select` transform to this inline function yields about a 50x speed improvement (2.2 seconds => 0.042 seconds). While this is a highly optimised function, it does somewhat raise concerns around the ongoing performance of the `select` transform on large datasets...